### PR TITLE
feat: add `allowPreselectedPaymentView` to Drop-in

### PR DESCRIPTION
**Pull request by descorp**
### Changes

- allow skipping `PreselectedPaymentMethod` view

### CODE REVIEW

File: Tests/AdyenTests/Adyen Tests/DropIn/DropInTests.swift [hunk 1]
```diff
@@ -168,6 +168,25 @@ class DropInTests: XCTestCase {
...
+
+        let paymentMethods = try! JSONDecoder().decode(PaymentMethods.self, from: DropInTests.paymentMethodsOneClick.data(using: .utf8)!)
+        sut = DropInComponent(paymentMethods: paymentMethods, configuration: config)
+
+        let root = UIViewController()
+        UIApplication.shared.keyWindow?.rootViewController = root
+        root.present(sut.viewController, animated: true, completion: nil)
+
+        let waitExpectation = expectation(description: "Expect DropIn to open")
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + .seconds(2)) {
```
Comment (mohammedDehairy) [inline Tests/AdyenTests/Adyen Tests/DropIn/DropInTests.swift:182]: you can just use `wait(for: .seconds(2))`

File: AdyenDropIn/DropInComponent.swift [hunk 2]
```diff
@@ -158,7 +158,8 @@ public final class DropInComponent: NSObject, PresentableComponent {
     }
     
     internal lazy var rootComponent: PresentableComponent = {
-        if let preselectedComponents = componentManager.storedComponents.first {
+        if configuration.allowPreselectedPaymentView,
+           let preselectedComponents = componentManager.storedComponents.first {
```
Comment (mohammedDehairy) [inline AdyenDropIn/DropInComponent.swift:162]: but then if `configuration.allowPreselectedPaymentView` is false, you need to show all stored payment methods in the second payment method list, right?

Comment (descorp) [reply AdyenDropIn/DropInComponent.swift:162]: Of cause, all stored payment methods are there:
This only allows to skip `PreselectedPaymentMethodComponent` and goes directly to a `PaymentMethodListComponent`.

File: Tests/AdyenTests/Adyen Tests/DropIn/DropInTests.swift (hunk 1 above)
Comment (descorp) [reply Tests/AdyenTests/Adyen Tests/DropIn/DropInTests.swift]: 👍

**Review by mohammedDehairy — APPROVED** [review]

**Review by nauaros — APPROVED** [review]
Nice!

**Pull request merged**